### PR TITLE
Added realtime-purge to publish command

### PR DESCRIPTION
**Pull request by askazion**

### CODE REVIEW

File: pkg/cmd/publish/publish.go [hunk 1]
```diff
@@ -185,16 +186,42 @@ func (cmd *publishCmd) run(f *cmdutil.Factory, info *publishInfo, options *contr
 		}
 	}
 
-	fmt.Fprintf(cmd.f.IOStreams.Out, "\nYour Domain name: %s\n", domain.GetDomainName())
+	var domainReturnedName []string
+	domainReturnedName = append(domainReturnedName, domain.GetDomainName())
```
Comment (fmiamoto) [inline pkg/cmd/publish/publish.go:190]: A more direct way to this do this is to initialize the slice directly
```suggestion
	domainReturnedName := []string{domain.GetDomainName()}
```

File: pkg/cmd/publish/publish.go [hunk 2]
```diff
@@ -185,16 +186,42 @@ func (cmd *publishCmd) run(f *cmdutil.Factory, info *publishInfo, options *contr
...
+		return err
+	}
+
+	if pg.StatusCode == 201 {
+		fmt.Fprintf(cmd.f.IOStreams.Out, "\nDomain cache was purged\n")
+		return nil
+	} else {
+		return fmt.Errorf("%s: %w", errmsg.ErrorPurgeDomainCache, err)
+	}
+}
```
Comment (fmiamoto) [inline pkg/cmd/publish/publish.go:223]: You can avoid the else here
```suggestion
	if pg.StatusCode != 201 {
	        return fmt.Errorf("%s: %w", errmsg.ErrorPurgeDomainCache, err)
	}
        fmt.Fprintf(cmd.f.IOStreams.Out, "\nDomain cache was purged\n")
        return nil
}
```
https://github.com/uber-go/guide/blob/master/style.md#unnecessary-else

File: pkg/api/realtime_purge/realtime_purge.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+)
+
+type Client struct {
+	apiClient *sdk.APIClient
+}
+
+type DomainResponse interface {
+	GetId() int64
+	GetDomainName() string
+}
```
Comment (fmiamoto) [inline pkg/api/realtime_purge/realtime_purge.go:20]: I think we're not using this interface

Comment (askazion) [reply pkg/api/realtime_purge/realtime_purge.go:20]: sure, removed

File: pkg/api/realtime_purge/realtime_purge.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	conf.Servers = sdk.ServerConfigurations{
+		{URL: url},
+	}
+
+	return &Client{
+		apiClient: sdk.NewAPIClient(conf),
+	}
+}
+
+func (c *Client) Purge(ctx context.Context, urlToPurge *[]string) (*http.Response, error) {
```
[the remark this answers is not in the document]
Comment (askazion) [reply pkg/api/realtime_purge/realtime_purge.go:37]: done

File: pkg/cmd/publish/publish.go (hunk 1 above)
Comment (askazion) [reply pkg/cmd/publish/publish.go:190]: done

File: pkg/cmd/publish/publish.go [hunk 5]
```diff
@@ -185,16 +186,42 @@ func (cmd *publishCmd) run(f *cmdutil.Factory, info *publishInfo, options *contr
...
+		err = cmd.purgeDomain(f, domainReturnedName)
+		if err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+func (cmd *publishCmd) purgeDomain(f *cmdutil.Factory, domainReturnedName []string) error {
```
[the remark this answers is not in the document]
Comment (askazion) [reply pkg/cmd/publish/publish.go:209]: done

[the remark this answers is not in the document]
Comment (askazion) [reply pkg/cmd/publish/publish.go:209]: done

File: pkg/cmd/publish/publish.go (hunk 2 above)
Comment (askazion) [reply pkg/cmd/publish/publish.go:223]: done

File: pkg/api/realtime_purge/realtime_purge.go [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+	if err != nil {
+		responseBody, _ := ioutil.ReadAll(httpRes.Body)
+		return fmt.Errorf("%w: %s", err, responseBody)
+	}
+
+	if httpRes.StatusCode != 201 {
+		return fmt.Errorf("%w: %s", err, httpRes.Status)
+	}
+
+	fmt.Println("\nDomain cache was purged")
```
Comment (fmiamoto) [inline pkg/api/realtime_purge/realtime_purge.go:48]: I'd suggest moving this Println to the Publish method itself.
There we can use whatever output was provided by the factory to print:
Something like:
```
		err = cmd.purgeDomains(f, domainReturnedName)
		if err != nil {
			return err
		}
                fmt.Fprintln(cmd.f.IOStreams.Out, "\nDomain cache was purged") // Added 
```

Comment (askazion) [reply pkg/api/realtime_purge/realtime_purge.go:48]: done

**Review by fmiamoto — APPROVED** [review]

**Pull request merged**